pkg/templates/templates_suite_test.go

```diff
@@ -0,0 +1,118 @@
+package templates
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+const testNs = "testns"
+
+var (
+	k8sConfig *rest.Config
+	k8sClient kubernetes.Interface
+	testEnv   *envtest.Environment
+	ctx       context.Context
+	cancel    context.CancelFunc
+)
+
+func TestMain(m *testing.M) {
+	os.Exit(testMain(m))
+}
+
+func testMain(m *testing.M) int {
+	setUp()
+
+	defer tearDown()
```

Comment:
Could you please put the `defer` before `setUp`? That way if `setUp` panics, `tearDown` still gets executed.